### PR TITLE
More support for java test suites.

### PR DESCRIPTION
**Pull request by tjoneslo**
This makes two changes to support further use of the Java Test suites matching what the junit5 testing allows. To that end there are two small changes here:
- expose the create_jvm_test_suite() macro to have access to set the define_test() for duplicate name processing.
- set the create_lint_tests() macro to avoid creating duplicate linter tests.

### CODE REVIEW

**Review by shs96c — APPROVED** [review]

File: java/private/library.bzl [hunk 1]
```diff
@@ -37,7 +37,7 @@ def create_lint_tests(name, **kwargs):
         )
 
     spotbugs = get_lint_config("java-spotbugs", tags)
-    if spotbugs != None:
+    if spotbugs != None and not native.existing_rule("%s-spotbuts" % name):
```
Comment (illicitonion) [inline java/private/library.bzl:40]:
```suggestion
    if spotbugs != None and not native.existing_rule("%s-spotbugs" % name):
```

Comment (illicitonion) [reply java/private/library.bzl:40]: Would it make more sense to use a `maybe` rather than hard-coding its impl here in ways which could contain typos?

Comment (tjoneslo) [reply java/private/library.bzl]: `maybe()` is a repository rule, and this isn't loading a repository. Good thought though.

Comment (illicitonion) [reply java/private/library.bzl]: There's actually nothing repository-rule specific about `maybe`, it works everywhere!

**Review by illicitonion — APPROVED** [review]
Thanks!

**Pull request merged**